R/pkg/inst/worker/worker.R

```diff
@@ -181,17 +185,35 @@ if (isEmpty != 0) {
                     colNames, computeFunc, data)
        } else {
         # gapply mode
+        outputs <- list()
         for (i in 1:length(data)) {
           # Timing reading input data for execution
           inputElap <- elapsedSecs()
           output <- compute(mode, partition, serializer, deserializer, keys[[i]],
                       colNames, computeFunc, data[[i]])
           computeElap <- elapsedSecs()
-          outputResult(serializer, output, outputCon)
+          if (deserializer == "arrow") {
+            requireNamespace("arrow", quietly = TRUE)
+            outputs[[length(outputs) + 1L]] <- output
+          } else {
+            outputResult(serializer, output, outputCon)
+          }
           outputElap <- elapsedSecs()
           computeInputElapsDiff <-  computeInputElapsDiff + (computeElap - inputElap)
           outputComputeElapsDiff <- outputComputeElapsDiff + (outputElap - computeElap)
         }
+
+        if (deserializer == "arrow") {
+          requireNamespace("arrow", quietly = TRUE)
+          # See https://stat.ethz.ch/pipermail/r-help/2010-September/252046.html
+          # rbind.fill might be an anternative to make it faster if plyr is installed.
+          combined <- do.call("rbind", outputs)
```

Comment:
Actually, according to the link above, the fastest way is convert them as matrix. I tried it before but looks it's not able to keep its types apparently. I tried other ways too but this seems to be the only way.

Reply:
shouldn't `rbind(outputs)` work here?

Reply:
and be aware of string columns https://github.com/apache/spark/blob/master/R/pkg/inst/worker/worker.R#L35

Reply:
Seems `rbind(iris(dataframe, ...)` not working .. :(. 

Yea, in terms of strings as factors ..  for input, it should be okay about `stringsAsFactors` because it's directly converted from Arrow batch.

For output, yes, I think there might be a concern about strings as factors. I currently assume the given function should always return the same `data.frame`s to combine. So I guess it's okay. I would like to leave this concern separate if you don't mind. We don't take care of factors in JVM side anyway.

Reply:
shouldn't we turn it off (as `stringsAsFactors = FALSE`)? 

If I recall, we don't handle factor to JVM correctly, and compute() from the user function doesn't **really** have to return a data.frame per se.

Reply:
Yup, turning off should be okay too. Let me turn off then.

Reply:
Actually, setting `stringsAsFactors` to `FALSE` downgrades its performance roughly twice ( from 4 secs to 8 secs for the benchmark in the PR description). Since it's being ignored in JVM, shouldn't be a matter to leave it.